### PR TITLE
Function template

### PR DESCRIPTION
**Pull request by ltaylor16**

### CODE REVIEW

File: shroud/typemap.py [hunk 1]
```diff
@@ -145,6 +157,8 @@ def __init__(self, name, **kw):
         #            setattr(self, key, defvalue)
         self.__dict__.update(self.defaults)  # set all default values
         self.update(kw)
+        if not self.flat_name:
+            self.flat_name = flatten_name(self.cxx_type or "JJJTypemap")
```
Comment (kennyweiss) [inline shroud/typemap.py]: ``JJJTypemap`` ?

Comment (ltaylor16) [reply shroud/typemap.py]: During testing cxx_type was None for some types.   JJJTypemap will then show up in the output helping me identify how that type is being used.  So it's more of a debugging aid.

Comment (ltaylor16) [reply shroud/typemap.py]: I tracked down why cxx_type was None in some cases.  I committed fixes which properly sets flat_name for typedefs, enums, and template parameters.

Comment (kennyweiss) [reply shroud/typemap.py]: I figured it was a debugging artifact. Thanks for looking into it.

**Pull request merged**